cpp/source/deal.II/geometry.h

```diff
@@ -120,6 +121,22 @@ class Geometry
       std::shared_ptr<boost::property_tree::ptree> database);
 
   /**
+   * Merge a triangulation that is repeated multiple times with the current
+   * triangulation.
+   */
+  void merge_repetition(dealii::distributed::Triangulation<dim> &repetition,
+                        double offset);
+
+  /**
+   * Compute the length of the triangulation along @dimension. Cells which have
+   * material_id in the set @excluded_materials are skipped.
+   */
+  double compute_length(
+      dealii::distributed::Triangulation<dim> const &triangulation,
+      unsigned int dimension,
+      std::unordered_set<dealii::types::material_id> const &excluded_materials);
```

Comment:
Here is seems that you only use the x-direction. If this was used for more directions I would just compute the bounding box.